### PR TITLE
HARP-13852 Increase precision, otherwise tilt is too inaccurate

### PR DESCRIPTION
**Pull request by nzjony**
- Improve the epsilon used when comparing the dot product.
- When the tilt animation has finished, a final check is done to see if the targeted tilt is reached, and if not, we apply the delta needed to reach the desired tilt.

### CODE REVIEW

File: @here/harp-map-controls/lib/MapControls.ts [hunk 1]
```diff
@@ -628,6 +628,22 @@ export class MapControls extends EventDispatcher {
...
+                        MapViewUtils.orbitAroundScreenPoint(
+                            this.mapView,
+                            0,
+                            0,
+                            0,
+                            deltaAngle,
+                            this.m_maxTiltAngle
+                        );
+                        this.updateMapView();
+                    }
```
Comment (atomicsulfate) [inline @here/harp-map-controls/lib/MapControls.ts:646]: I think the call to this.stopTilt() is done to soon. It sets:
 this.m_targetedTilt = this.m_currentTilt = undefined;

Then later we pass to easeOutCubic() as target value this.targetedTilt, which returns the current mapView.tilt if this.m_targetedTilt  is undefined.

If the stopTilt() call is done after easOutCubic, together with your epsilon changes, it works fine for me.

Comment (atomicsulfate) [inline @here/harp-map-controls/lib/MapControls.ts:646]: I think the call to this.stopTilt() is done to soon. It sets:
this.m_targetedTilt = this.m_currentTilt = undefined;

Then later we pass to easeOutCubic() as target value this.targetedTilt, which returns the current mapView.tilt if this.m_targetedTilt is undefined.

If the stopTilt() call is done after easOutCubic, together with your epsilon changes, it works fine for me.

Comment (nzjony) [reply @here/harp-map-controls/lib/MapControls.ts:646]: Thanks @atomicsulfate. I have pushed a fix.

**Review by atomicsulfate — DISMISSED** [review]
Looks good. It'd be even better if we add a unit test to check we actually reach 0 tilt.

**Comment by nzjony** [comment]
@atomicsulfate , I agree, I will put the test in a separate PR, so we can at least get this into the release.

**Comment by nzjony** [comment]
retest this please

**Review by atomicsulfate — APPROVED** [review]

**Pull request merged**